samples/nrf9160/lwm2m_client/src/lwm2m/lwm2m_location.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2019 Foundries.io
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#define LOG_MODULE_NAME app_lwm2m_loc
+#define LOG_LEVEL CONFIG_APP_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <zephyr.h>
+#include <gps.h>
+#include <stdio.h>
+#include <net/lwm2m.h>
+
+#include "ui.h"
+
+static struct gps_data nmea_data;
+static struct device *gps_dev;
+static struct gps_trigger gps_trig = {
+	.type = GPS_TRIG_DATA_READY,
+};
+
+/*
+ * @brief Helper function to convert from double to sensor_value struct
+ *
+ * @param val Sensor value to convert.
+ * @param sense_val Pointer to sensor_value to store the converted data.
+ */
+static void double_to_float32_value(double val,
+				    struct float32_value *f32_val)
+{
+	f32_val->val1 = (int)val;
+	f32_val->val2 = (val - (int)val) * 1000000;
+}
+
+static void update_location_data(struct gps_data *nmea)
+{
+	LOG_DBG("nmea_data");
```

Comment:
@joakimtoe Do you know if GPS socket provides us with NMEA string, or a structure with separate fields for latitude, longitude etc.? LWM2M expects to have these values in separate fields instead of being encoded in the NMEA string.

Reply:
@rlubos @joakimtoe From the GPS sample, it looks like if the modem is in GPS mode (XSYSTEMMODE=0,0,1,0) then I can use `nrf_recv` to fill a `nrf_gnss_data_frame_t structure`.  This structure contains `nrf_gnss_pvt_data_frame_t` which has longitude, latitude, etc.  That would be the best data format for the Location object in the LwM2M client.

I have to wonder if both LTE and GPS can be used at the same time?  There is no sample which shows that currently.

Reply:
@mike-scott There's a limitation on GPS use - the modem needs to be in power saving mode in order to get measurements. From the nRF9160 Product Specification:
```
The GPS receiver supports GPS L1 C/A reception. Operation is time multiplexed with LTE 
modem, and it is possible to obtain the GPS position either while the LTE is in RRC Idle
mode or power saving mode (PSM), or when the LTE modem is completely deactivated.
```
https://infocenter.nordicsemi.com/topic/ps_nrf9160/gps.html